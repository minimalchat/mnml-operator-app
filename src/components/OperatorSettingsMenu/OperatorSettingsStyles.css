.SettingsMenu {
  width: 100%;
}

.Settings__box {
  display: flex;
  background:  #F7F8FA;
  border-top: 1px solid  #E8ECEE;
  border-bottom: 1px solid  #E8ECEE;
  justify-content: flex-start;
  width: 100%;
  font-size: 12px;
  color: #8f8f8f;
  align-items: center;
}

.Settings__avatar {
  margin: 1rem;
  border-radius: 50%;
  border: 3px solid white;
  box-shadow: 1px 1px 0 rgba(0, 0, 0, 0.15);
  width: 58px;
  height: 58px;
}

.Settings__operator {
  word-break: break-word;
}
