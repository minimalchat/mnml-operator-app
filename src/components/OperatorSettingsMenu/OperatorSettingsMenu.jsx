/** OperatorSettingsMenu
  * Displays informaiton about the operator.
  * Might eventually hold a menu settings button?
*/

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './OperatorSettingsStyles.css';

const OperatorSettingsMenu = ({ avatar, operatorName }) => (
  <section className="SettingsMenu">
    <div className="Settings__box">
      <div className="Settings__avatar">
        <img
          alt={operatorName}
          src={avatar ? `https:${avatar}` : 'https://www.fillmurray.com/58/58'}
        />
      </div>
      <span className="Settings__operator">{operatorName}</span>
    </div>
  </section>
);

OperatorSettingsMenu.propTypes = {
  operatorName: PropTypes.string,
  avatar: PropTypes.string,
};

OperatorSettingsMenu.defaultProps = {
  operatorName: '',
  avatar: null,
};

const mapStateToProps = state => ({
  operatorName: state.config.operator,
  avatar: state.config.avatar,
});

export default connect(
  mapStateToProps,
)(OperatorSettingsMenu);
