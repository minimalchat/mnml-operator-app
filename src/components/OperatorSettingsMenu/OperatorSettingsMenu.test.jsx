import React from 'react';

import { shallow } from 'enzyme';

import OperatorSettingsMenu from './OperatorSettingsMenu.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ config: {} })),
};

describe('OperatorSettingsMenu', () => {
  it('matches snapshot', () => {
    const component = shallow(<OperatorSettingsMenu store={store} />);

    expect(component).toMatchSnapshot();
  });
});
