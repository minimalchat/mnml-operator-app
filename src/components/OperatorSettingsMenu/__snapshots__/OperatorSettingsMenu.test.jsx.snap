// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorSettingsMenu matches snapshot 1`] = `
<OperatorSettingsMenu
  avatar={null}
  dispatch={[MockFunction]}
  operatorName=""
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Object {
              "config": Object {},
            },
          },
          Object {
            "isThrow": false,
            "value": Object {
              "config": Object {},
            },
          },
        ],
      },
      "subscribe": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
    }
  }
/>
`;
